tempest/api/workloadmgr/base.py

```diff
@@ -3176,6 +3176,8 @@ def workload_snapshot_cli(self, workload_id, is_full):
     '''
 
     def create_restore_json(self, rest_details):
+        if 'volume_type' in rest_details.keys():
```

Comment:
In test_multiattach_volumes.py --> selective restore part - you are updating rest_details as,
rest_details['volume_type'] = vol_type_name

Now here you are checking if volume_type present in rest_details as a key or not. If it is present then you are overwriting that value with "CONF.volume.volume_type" . It means, the value passed inside rest_details to create_restore_json() method is getting overwritten.

Please check, if you wanted to write like,
if 'volume_type' not in rest_details.keys():
rest_details['volume_type'] = CONF.volume.volume_type

otherwise the value passed in rest_details will remain as it is.